src/inmanta/data/__init__.py

```diff
@@ -5390,14 +5391,42 @@ async def create_for_partial_compile(
                 ) AS base_version_found
             ),
             rids_undeployable_base_version AS (
-                SELECT DISTINCT unnest(c2.undeployable) AS rid
-                FROM {cls.table_name()} AS c2
-                WHERE c2.environment=$1 AND c2.version=$8
+                SELECT t.rid
+                FROM (
+                    SELECT DISTINCT unnest(c2.undeployable) AS rid
+                    FROM {cls.table_name()} AS c2
+                    WHERE c2.environment=$1 AND c2.version=$8
+                ) AS t(rid)
+                WHERE (
+                    EXISTS (
+                        SELECT 1
```

Comment:
This is equivalent to a `JOIN` because it selects on (stricter than) the primary key? I'd find `JOIN` a bit more readable, but fine by me if you prefer it like this.